lib/community/pages/create_post_page.dart

```diff
@@ -29,54 +33,48 @@ class CreatePostPage extends StatefulWidget {
 
 class _CreatePostPageState extends State<CreatePostPage> {
   bool showPreview = false;
-  bool isClearButtonDisabled = false;
   bool isSubmitButtonDisabled = true;
   bool isNSFW = false;
-  bool error = false;
+  String url = "";
 
-  // final List<bool> _selectedPostType = <bool>[true, false, false];
-  String image = '';
-  String description = '';
   final TextEditingController _bodyTextController = TextEditingController();
   final TextEditingController _titleTextController = TextEditingController();
+  final TextEditingController _urlTextController = TextEditingController();
+  final FocusNode _markdownFocusNode = FocusNode();
   final imageBloc = ImageBloc();
 
   @override
   void initState() {
     super.initState();
 
-    _bodyTextController.addListener(() {
-      if (_bodyTextController.text.isEmpty && !isClearButtonDisabled) setState(() => isClearButtonDisabled = true);
-      if (_bodyTextController.text.isNotEmpty && isClearButtonDisabled) setState(() => isClearButtonDisabled = false);
-    });
-
     _titleTextController.addListener(() {
       if (_titleTextController.text.isEmpty && !isSubmitButtonDisabled) setState(() => isSubmitButtonDisabled = true);
       if (_titleTextController.text.isNotEmpty && isSubmitButtonDisabled) setState(() => isSubmitButtonDisabled = false);
     });
+
+    _urlTextController.addListener(() {
+      if (url != _urlTextController.text) {
+        setState(() {
+          url = _urlTextController.text;
```

Comment:
This state change triggers for every character input. This causes a UI redraw for the owning widget and children on every keystroke in the URL text box. 

Maybe some sort of debounce type feature would be a good idea. Where we don't change state and check for the existing URL until the user has stopped typing for 'n' amount of time. @hjiangsu Thoughts?

Reply:
Yup, I agree with @ajsosa here. There should be a util which you can use for the debounce logic in `lib/utils/debounce.dart`!

Reply:
Added a one second delay before `setState`.